onnxruntime/core/providers/webnn/builders/model.cc

```diff
@@ -26,6 +26,16 @@ Model::~Model() {}
 
 Status Model::Predict(const InlinedHashMap<std::string, OnnxTensorData>& inputs,
                       const InlinedHashMap<std::string, OnnxTensorData>& outputs) {
+  if (webnn::IsMlBufferSupported()) {
```

Comment:
I check latest Chrome Stable, `MLBuffer` is already defined, thus it will always return `Dispatch()`.

But the MLBuffer features are not fully supported, e.g. Running on CPU backend on latest Canary, it will throws: 
`NotSupportedError: Failed to execute 'writeBuffer' on 'MLContext': Not implemented`.

Looks like at current stage, we can't tell if MLBuffer is supported for specific backends or browsers. We may need to hold this PR at least until all MLBuffer features be implemented in all backends on latest Chrome Canary.

cc/ @huningxin

Reply:
@Honry CPU backend is not fully supported, dispatch() remains unimplemented. There are no plans to to do feature detection for `MLBuffer` so I worry holding this PR could delay us further testing the GPU backend.

Can we disable MLBuffer and keep using compute() when using the CPU, in the meantime?

@huningxin, a good issue for us to raise with Google on TFLite support.

Reply:
I'm fine to disable `MLBuffer` for CPU backend for now.

Reply:
It seems kind unfortunate to complicate the current `IsMlBufferSupported` implementation for a temporary workaround. Do we know the timeline for the TFLite backend support for `dispatch`?

Reply:
Do we know what's the plan of MLBuffer / dispatch() support on WebNN TFLite / CoreML backends? @reillyeon @a-sully

Reply:
I'm working on `MLBuffer` / `dispatch()` TFLite and @a-sully is working in it for Core ML. The change @bbernhar is considering to make `createBuffer()` return a `Promise` should make feature detection for this possible as well. However I don't think it should ever be optional.

Reply:
Added `deviceType` to `IsMlBufferSupported` and piped it around. We'll need to remove all this code once the CPU backend works and `compute` gets removed from the specification.